### PR TITLE
Metrics for `GraphBuilder`

### PR DESCRIPTION
**Pull request by rutujasurve94**
Add metrics for `GraphBuilder`.

### CODE REVIEW

File: src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  * {s}`GraphBuilderTelemetry`
+  *  : Adds metrics for {s}`GraphBuilder` using Prometheus Client
+  *
+  *    Exposes Counter and Gauge stats for tracking number of vertices and edges added and deleted, total partitions created
+  *    by the graph builder
+  *    Statistics are made available on http://localhost:8899 on running tests and can be visualised using Grafana dashboards
+  *
+  *
+  */
+object GraphBuilderTelemetry {
```
Comment (miratepuffin) [inline src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala:16]: Not sure if the names here are needed to add in the builderID. Might be an idea so we can see if one is lagging behind etc.

File: src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  val totalGraphBuilderUpdates: Counter =
+    Counter
+      .build
+      .namespace(conf.getString("raphtory.prometheus.builder_namespace"))
+      .name("graph_builder_total_updates")
+      .help("Total graph builder updates")
+      .register
+
+  val totalBuilderPartitions: Gauge =
```
Comment (miratepuffin) [inline src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala:60]: Can be removed I think

**Review by miratepuffin — CHANGES_REQUESTED** [review]

File: src/main/scala/com/raphtory/core/components/graphbuilder/GraphBuilder.scala [hunk 3]
```diff
@@ -176,6 +178,7 @@ trait GraphBuilder[T] extends Serializable {
     batchWriters = writers
     batching = true
     totalPartitions = partitions
+    GraphBuilderTelemetry.totalBuilderPartitions.set(partitions)
```
[the remark this answers is not in the document]
Comment (rutujasurve94) [reply src/main/scala/com/raphtory/core/components/graphbuilder/GraphBuilder.scala]: Added this for convenience - to extract `application.conf` properties out as metrics, if helpful. I'm removing this, if redundant

File: src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala (hunk 1 above)
Comment (rutujasurve94) [reply src/main/scala/com/raphtory/core/config/telemetry/GraphBuilderTelemetry.scala:16]: I've added a `builder_id` var to this object that could then be set as required.

**Comment by miratepuffin** [comment]
#297 #301 #302 and #339 to be merged into one PR off of the new Development branch so they can bee all tested together and merged in.

**Pull request closed without merging**